doobie/src/it/scala/za/co/absa/fadb/doobie/DatesTimesTest.scala

```diff
@@ -46,30 +44,24 @@ class DatesTimesTest extends AnyFunSuite with DoobieTest {
                              )
 
   class GetAllDateTimeTypes(implicit schema: DBSchema, dbEngine: DoobieEngine[IO])
-      extends DoobieSingleResultFunction[Int, DatesTimes, IO] {
-
-    override def sql(values: Int)(implicit read: Read[DatesTimes]): Fragment =
-      sql"SELECT * FROM ${Fragment.const(functionName)}($values)"
-  }
+      extends DoobieSingleResultFunction[Int, DatesTimes, IO](values => Seq(fr"$values"))
 
   class InsertDatesTimes(implicit schema: DBSchema, dbEngine: DoobieEngine[IO])
-      extends DoobieSingleResultFunctionWithStatus[DatesTimes, Int, IO] with StandardStatusHandling {
-
-    override def sql(values: DatesTimes)(implicit read: Read[StatusWithData[Int]]): Fragment =
-      sql"""
-           SELECT * FROM ${Fragment.const(functionName)}(
-            ${values.offsetDateTime},
-            ${values.instant},
-            ${values.zonedDateTime},
-            ${values.localDateTime},
-            ${values.localDate},
-            ${values.localTime},
-            ${values.sqlDate},
-            ${values.sqlTime},
-            ${values.sqlTimestamp},
-            ${values.utilDate}
-           )
-        """
+      extends DoobieSingleResultFunctionWithStatus[DatesTimes, Int, IO] (
+        values => Seq(
+          fr"${values.offsetDateTime}",
+          fr"${values.instant}",
+          fr"${values.zonedDateTime}",
+          fr"${values.localDateTime}",
+          fr"${values.localDate}",
+          fr"${values.localTime}",
+          fr"${values.sqlDate}",
+          fr"${values.sqlTime}",
+          fr"${values.sqlTimestamp}",
+          fr"${values.utilDate}"
+        )
+      ) with StandardStatusHandling {
+    override def fieldsToSelect: Seq[String] = super.fieldsToSelect ++ Seq("o_id")
```

Comment:
`o_id` ?

Reply:
It's just a dummy sql function that returns o_id column. Don't worry about it. We should rather introduce some better designed tests for testing inserts and retrievals of data of different data types.

Reply:
yes, I actually found out shortly after I wrote it, apologies